### ISSUE TITLE
Apply bandwidth limit to file listing

### ISSUE THREAD

**Issue by aetonsi** [rclone/rclone#4395]
#### What is your current rclone version (output from `rclone version`)?
I tried with 2 versions:
```
rclone v1.50.2
- os/arch: windows/amd64
- go version: go1.13.4
```
```
rclone v1.51.0-151-gfc663d98-beta
- os/arch: windows/amd64
- go version: go1.14.1
```
#### command line switches used
```--verbose --copy-links --create-empty-src-dirs --retries 6 --retries-sleep 30s --stats 60s --stats-file-name-length 0 --buffer-size 16M --tpslimit 1 --transfers 1 --checkers=1 --bwlimit 5k```

I tried with many different `--bwlimit` values, even `512b`.

#### What problem are you are trying to solve?
rclone kills my internet connection due to exaggerated network activity during file listing.
It happens because i have a pretty bad connection, and i have thousands of files synchronized on different cloud services.
So, whenever i run my script to `sync` between my local folders and the cloud folders, even if there are no files changed, rclone still saturates my bandwidth to do the file listing.
Internet consequently becomes unusable for everyone connected to my access point for a few seconds. This happens whenever i have a "full on" upload, it's not limited to rclone.

Briefly: not having a `bwlimit` on the file listing is deleterious, in case you have a weak internet connection.
It probably does not make any difference if you have a good connection.

#### How do you think rclone should be changed to solve that?
The switch `--bwlimit` could be applied also to the file listing.
As an alternative, a switch like `--filelistingbwlimit` could be added.
As an alternative, a switch like `--applybwlimittofilelisting` could be added to be used alongside `--bwlimit`

#### Example screenshots
![image](https://user-images.githubusercontent.com/18366087/86033864-ef5d6900-ba39-11ea-9068-3b7ba7d3a847.png)

**Comment by ncw** [comment]
You are right, listing bandwidth is not accounted for by bwlimit.

I'd need to put the limit in at a lower level, in the http transport. This wouldn't affect SFTP but it would catch most of the backends.

I would have thought your tpslimit would work reasonable well but I guess it doesn't ?

**Comment by aetonsi** [comment]
Well, even with uber-limiting switches ( `--tpslimit 1 --transfers 1 --checkers=1 --bwlimit 512b` ), the problem is very much present... 

I can tell whenever my sync script (running in background once per hour) is listing files on remotes, because all web browsing stops for 10 to 30 seconds, for me and everyone connected to the access point, and pinging a dns server (google's 8.8.8.8, always running in the background) reports reply times up to 3-4000ms.
Then (ironically), the "real" file upload starts and lagging stops (because of the `bwlimit` that i imposed, obviously).

So yeah, if it would be possible to move the limit to the http transport level, that would be absolutely wonderful.

---

ps: I know that the bigger problem upstream is the poor quality of (my) internet connection, but unluckily there's no better alternative where i live. And this might be true for many others current and future users.
I am limited in using rclone by this uncontrollable factor, and it's a shame because it's such a nice software

**Comment by ncw** [comment]
I'm due to be looking at the bandwidth limiting this dev cycle so I'll try to take a look then. Unless you want to help?

**Comment by aetonsi** [comment]
Sorry, i currently don't have the knowledge necessary to even know where to look

**Comment by ncw** [comment]
> Sorry, i currently don't have the knowledge necessary to even know where to look

No problems! I'll try to fit this in into the 1.53 dev cycle :crossed_fingers:

**Comment by aetonsi** [comment]
Thank you, kind sir

**Comment by ncw** [comment]
I've had a go at this - can you try it out please?

[v1.52.2-198-g6a399560-fix-4395-list-bwlimit-beta](https://beta.rclone.org/branch/v1.52.2-198-g6a399560-fix-4395-list-bwlimit-beta/) on branch [fix-4395-list-bwlimit](https://github.com/rclone/rclone/tree/fix-4395-list-bwlimit) (uploaded in 15-30 mins)

**Comment by aetonsi** [comment]
Hi, i'm on windows and i don't see a windows build. Is it correct? Should i clone it from the repo and build it, instead?

**Comment by ncw** [comment]
Oops, sorry, broke the build. Try this one

[v1.52.2-198-gaedc8623-fix-4395-list-bwlimit-beta](https://beta.rclone.org/branch/v1.52.2-198-gaedc8623-fix-4395-list-bwlimit-beta/) on branch [fix-4395-list-bwlimit](https://github.com/rclone/rclone/tree/fix-4395-list-bwlimit) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
@aetonsi did that build work for you?

**Comment by aetonsi** [comment]
@ncw hello, sorry for late reply, it seems to work.
I used `--dry-run` to test the file listing only. 

- with ` --bwlimit 512b` (which is my usual value), file listing did not impact my connection and took around 1 minute
- without the bwlimit, file listing impacted my connection negatively and took around 20 seconds less, for the same files

So i'd say it works.
Tell me if you need any other testing done.

Thanks!

**Comment by ncw** [comment]
That is great thank you!

I'm not going to merge this just yet as I want to combine it with the other work on setting independent upload and download limits

Hopefully it will be in 1.53

**Comment by ncw** [comment]
Here is the next iteration of this

[v1.54.0-beta.4959.d7c84b094.fix-4395-list-bwlimit](https://beta.rclone.org/branch/fix-4395-list-bwlimit/v1.54.0-beta.4959.d7c84b094.fix-4395-list-bwlimit/) on branch [fix-4395-list-bwlimit](https://github.com/rclone/rclone/tree/fix-4395-list-bwlimit) (uploaded in 15-30 mins)

Whenever you write a bandwidth `--bwlimit 100k` you can now write a pair `--bwlimit 10M:1M` which is upload:download.

**Comment by aetonsi** [comment]
> Here is the next iteration of this
> 
> [v1.54.0-beta.4959.d7c84b094.fix-4395-list-bwlimit](https://beta.rclone.org/branch/fix-4395-list-bwlimit/v1.54.0-beta.4959.d7c84b094.fix-4395-list-bwlimit/) on branch [fix-4395-list-bwlimit](https://github.com/rclone/rclone/tree/fix-4395-list-bwlimit) (uploaded in 15-30 mins)
> 
> Whenever you write a bandwidth `--bwlimit 100k` you can now write a pair `--bwlimit 10M:1M` which is upload:download.

Cool!

Would it be possible to specify just an upload/download limit with, for example, some special value to mean "no limit"? like this: `--bwlimit 50k:inf` or just `--bwlimit 50k:`

Or to be cleaner: `--bwlimit 50k:2m`  /  `--bwlimit-download 50k`  /  `--bwlimit-upload 50k`

**Comment by ncw** [comment]
@aetonsi wrote:
> Would it be possible to specify just an upload/download limit with, for example, some special value to mean "no limit"? like this: `--bwlimit 50k:inf` or just `--bwlimit 50k:`
> 
> Or to be cleaner: `--bwlimit 50k:2m` / `--bwlimit-download 50k` / `--bwlimit-upload 50k`

`off` should work I think

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.54
